dfu_gecko/dfu_gecko_lib.h

```diff
@@ -16,25 +16,19 @@
 #define DFU_SHA1_LEN 20
 #define DFU_URL_LEN  256
 
-#define CONFIG_MCU_NAME "Pearl Gecko"
-
-#define LOG_INFO printf
-#define LOG_DEBUG printf
-#define LOG_ERROR printf
-#define LOG_WARN printf
-#define ENDL ""
-
 struct dfu_file_t {
 	char desc[DFU_DESC_LEN];
 	char lfile[DFU_FILE_LEN];
 	char rfile[DFU_FILE_LEN];
 	char sha1[DFU_SHA1_LEN];
 };
 
-//const struct dfu_file_t *dfu_file;
-
-int get_gecko_fw_version (void);
```

Comment:
Can we still keep this and have this return a char* of the version with the parameter boot slot?
char* get_gecko_fw_version (int boot_slot);

Reply:
i’d suggest this instead: int get_gecko_fw_version(int boot_slot, char *version, int max_len);
return would be 0 if ok or -1 if no version available

Reply:
Sounds good to me!